File: java-micronaut/src/main/java/com/claude/bakeoff/repository/UserRepository.java
package com.claude.bakeoff.repository;

import com.claude.bakeoff.model.User;
import com.mongodb.client.MongoClient;
import com.mongodb.client.MongoCollection;
import com.mongodb.client.MongoCursor;
import com.mongodb.client.model.Filters;
import jakarta.inject.Singleton;
import org.bson.Document;
import org.bson.types.ObjectId;

import java.time.Instant;
import java.util.ArrayList;
import java.util.List;
import java.util.Optional;

@Singleton
public class UserRepository {
    
    private final MongoCollection<Document> collection;
    
    public UserRepository(MongoClient mongoClient) {
        this.collection = mongoClient.getDatabase("bakeoff").getCollection("users");
    }
    
    public List<User> findAll() {
        List<User> users = new ArrayList<>();
        try (MongoCursor<Document> cursor = collection.find().iterator()) {
            while (cursor.hasNext()) {
                users.add(documentToUser(cursor.next()));
            }
        }
        return users;
    }
    
    public List<User> findByDepartment(String department) {
        List<User> users = new ArrayList<>();
        try (MongoCursor<Document> cursor = collection.find(Filters.eq("department", department)).iterator()) {
            while (cursor.hasNext()) {
                users.add(documentToUser(cursor.next()));
            }
        }
        return users;
    }
    
    public Optional<User> findByEmail(String email) {
        Document doc = collection.find(Filters.eq("email", email)).first();
        return doc != null ? Optional.of(documentToUser(doc)) : Optional.empty();
    }
    
    public long count() {
        return collection.countDocuments();
    }
    
    private User documentToUser(Document doc) {
        // Handle salary type conversion (might be Integer or Double)
        Object salaryValue = doc.get("salary");
        double salary = 0.0;
        if (salaryValue instanceof Integer) {
            salary = ((Integer) salaryValue).doubleValue();
        } else if (salaryValue instanceof Double) {
            salary = (Double) salaryValue;
        }
        
        return new User(
            doc.getObjectId("_id").toString(),
            doc.getString("name"),
            doc.getString("email"),
            doc.getInteger("age", 0),
            doc.getString("department"),
            salary,
            doc.getList("skills", String.class),
            doc.getDate("createdAt").toInstant()
        );
    }
}

File: kotlin-micronaut/src/main/kotlin/com/claude/bakeoff/repository/UserRepository.kt
package com.claude.bakeoff.repository

import com.claude.bakeoff.model.User
import com.mongodb.client.MongoClient
import com.mongodb.client.MongoCollection
import com.mongodb.client.model.Filters
import jakarta.inject.Singleton
import org.bson.Document

@Singleton
open class UserRepository(private val mongoClient: MongoClient) {
    
    private val collection: MongoCollection<Document> = 
        mongoClient.getDatabase("bakeoff").getCollection("users")
    
    open fun findAll(): List<User> {
        return collection.find().map { documentToUser(it) }.toList()
    }
    
    open fun findByDepartment(department: String): List<User> {
        return collection.find(Filters.eq("department", department))
            .map { documentToUser(it) }.toList()
    }
    
    open fun findByEmail(email: String): User? {
        return collection.find(Filters.eq("email", email)).first()?.let { documentToUser(it) }
    }
    
    open fun count(): Long {
        return collection.countDocuments()
    }
    
    private fun documentToUser(doc: Document): User {
        return User(
            id = doc.getObjectId("_id").toString(),
            name = doc.getString("name"),
            email = doc.getString("email"),
            age = doc.getInteger("age", 0),
            department = doc.getString("department"),
            salary = when (val salaryValue = doc.get("salary")) {
                is Int -> salaryValue.toDouble()
                is Double -> salaryValue
                else -> 0.0
            },
            skills = doc.getList("skills", String::class.java) ?: emptyList(),
            createdAt = doc.getDate("createdAt").toInstant()
        )
    }
}